src/fortran_routing/diffusive_v02frwk/diffusive_cnt/diffusive_cnt.f90

```diff
@@ -278,11 +278,47 @@ subroutine diffnw(dtini_g, t0_g, tfin_g, saveinterval_ev_g, dt_ql_g, dt_ub_g, dt
                 q_sk_multi=1.0
                 slope = (z(ncomp-1,j)-z(ncomp,j))/dx(ncomp-1,j)
                 if (slope .le. 0.0001) slope = 0.0001
+		oldQ(ncomp,j)= iniq(ncomp,j)
                 !oldY below takes normal depth value as a result.
                 call normal_crit_y(ncomp, j, q_sk_multi, slope, oldQ(ncomp,j), oldY(ncomp,j), temp,  oldArea(ncomp,j), temp)
             end if
         end do
         ! +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+        ! compute initial water depth, celerity, and diffusivity using iniq
+        ! +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+        q_sk_multi=1.0
+        do j=1, nlinks
```

Comment:
This do loop also happens above. Can we take care of all the initialization business with one do loop, rather than two? Seems repetitive.